### DISCUSSION TITLE
Concerns with Weblate's Statistic Calculation on Languages

### DISCUSSION THREAD

**Discussion by vanlinh2204** [WeblateOrg/weblate#11153] [Q&A]
The problem arises when observing the statistics in the category ipension-front-office. Here's a breakdown of the issue:

![image](https://github.com/WeblateOrg/weblate/assets/133736755/287396ee-4a0e-460e-bddb-6616f1dbc083)

Uniform Unfinished String Count: Within the ipension-front-office category, I've noticed that the unfinished string count remains consistent across 4 languages: fr, de, it, en. However, upon closer inspection, this count does not accurately reflect the number of strings requiring translation in each language. For instance, no strings appear for translation when selecting the French (fr) language, despite the reported count.

![image](https://github.com/WeblateOrg/weblate/assets/133736755/3b691dc5-b1ac-4267-9333-85f866a71a5b)

Correct Display in Sub-categories: Conversely, when navigating to the sub-category ipension-front-office/portal-service, the language tab correctly displays the number of unfinished strings, such as 40 in English (en). Further exploration into the components within ipension-front-office/portal-service, such as core-portal-service (10 unfinished in English) and first-portal-services (30 unfinished in English), indicates the accurate count of unfinished strings.

![image](https://github.com/WeblateOrg/weblate/assets/133736755/52938071-5ea4-44c0-94a3-77322e544f9b)
![image](https://github.com/WeblateOrg/weblate/assets/133736755/9c1a5c10-ae75-4412-b8ed-10ae8edfcf2f)
![image](https://github.com/WeblateOrg/weblate/assets/133736755/4dce269d-d6ef-4313-a87b-64dac6b2d601)

This inconsistency in statistic calculation poses challenges for translators and project managers relying on Weblate's data accuracy for localization efforts.

I kindly request your assistance in addressing this matter and welcome any suggestions or insights you may have regarding this issue. Additionally, if anyone has encountered similar challenges or has insights into potential solutions, please share them here.

Thank you for your attention to this matter, and I look forward to your responses.

**Comment by nijel** [comment]
We had some issues in updating language stats in past releases (see issues like https://github.com/WeblateOrg/weblate/issues/10540). Maybe upgrading would address your issues.

**Reply by vanlinh2204** [reply]
Hello @nijel,

I've also updated my Weblate version from 5.2.1 to 5.4.1, but it's not functioning properly.

After examining all components and categories within a single project, I've noticed the following:

If a category contains only components, the statistics display correctly. Clicking the link for unfinished languages redirects to the correct values.
![image](https://github.com/WeblateOrg/weblate/assets/133736755/d4678b16-b878-494a-91f7-d380a0219eb2)
![image](https://github.com/WeblateOrg/weblate/assets/133736755/58309522-02e2-495b-8ec5-c4c364b0a400)


However, when the category contains sub-categories, it doesn't display as expected. For instance, let's consider a category "ipension/commons" with two sub-categories (batch, commons-gui):
![image](https://github.com/WeblateOrg/weblate/assets/133736755/82311f56-704d-43cf-9977-b718aa531a17)

"batch" has 815 unfinished strings (fr=1, en=435, de=239, it=140)
![image](https://github.com/WeblateOrg/weblate/assets/133736755/0fd0c888-5f57-4a04-9e22-8e0991e85452)

"commons-gui" has 55 unfinished strings (fr=51, en=1, de=3)
![image](https://github.com/WeblateOrg/weblate/assets/133736755/be4d33ee-1042-459f-ae8c-0da8f2f5e3d0)


The sum of these is 815 + 55 = 870.

However, on the languages tab of the category "commons," it displays fr=de=it=en=870, which is incorrect because it's the sum of all languages. Thus, when clicking into the unfinished strings, it displays the wrong numbers.
![image](https://github.com/WeblateOrg/weblate/assets/133736755/1ea636d4-b997-4505-877e-c63875cc6a99)
![image](https://github.com/WeblateOrg/weblate/assets/133736755/e42375be-6911-4c7b-b7db-853966b9bc91)

It should display fr=52, en=436, de=242, it=140.

The other point is that the project also displays correctly.
![image](https://github.com/WeblateOrg/weblate/assets/133736755/814643c0-8c03-4cfe-b8fe-4608587ea37c)

**Reply by nijel** [reply]
Updating will not fix current statistics, these are updated only on change. Does the stats fix if you edit a string in given language?

**Reply by vanlinh2204** [reply]
Hello @nijel, despite making numerous changes, the issue persists with categories containing subcategories.

**Comment by nijel** [comment]
Turned this into https://github.com/WeblateOrg/weblate/issues/11240
